src/GafferOSL/OSLShader.cpp

```diff
@@ -131,6 +133,9 @@ ConstShadingEnginePtr getter( const ShadingEngineCacheGetterKey &key, size_t &co
 typedef IECorePreview::LRUCache<IECore::MurmurHash, ConstShadingEnginePtr, IECorePreview::LRUCachePolicy::Parallel, ShadingEngineCacheGetterKey> ShadingEngineCache;
 ShadingEngineCache g_shadingEngineCache( getter, 10000 );
 
+typedef tbb::concurrent_unordered_set<IECore::TypeId> ShaderTypeSet;
```

Comment:
I don't think thread safety is something we need here (see other comment about registering statically). The main feature of our usage of the container is that it will have very few entries and we will hardly every write to it, but we will read from it frequently. `boost::flat_set` is a decent choice in that scenario.

Reply:
Cheers for the suggestion, yeah I wasn't wanting to go the concurrent set if I didn't have to but your static bool check sounds like the way to go above, I'll fix this up.